osvr/RenderKit/RenderManagerD3D11ATW.h

```diff
@@ -528,62 +514,68 @@ namespace osvr {
+                    if (newInfo.textureCopy) {
+                      textureToMap = newInfo.textureCopy;
+                    }
+
+                    // we need to get the shared resource HANDLE for the ID3D11Texture2D, but in order to
+                    // get that, we need to get the IDXGIResource* first.
+                    ID3D11Texture2D *texture2D = nullptr;
+                    IDXGIResource* dxgiResource = nullptr;
+                    hr = textureToMap->QueryInterface(__uuidof(IDXGIResource), (LPVOID*)&dxgiResource);
+                    if (FAILED(hr)) {
+                      m_log->error()
+                        << "RenderManagerD3D11ATW::"
+                        << "RegisterRenderBuffersInternal: Can't get the IDXGIResource for the texture resource.";
+                      setDoingOkay(false);
+                      return false;
+                    }
+
+                    // now get the shared HANDLE
+                    hr = dxgiResource->GetSharedHandle(&newInfo.sharedResourceHandle);
+                    if (FAILED(hr)) {
+                      m_log->error()
+                        << "RenderManagerD3D11ATW::"
+                        << "RegisterRenderBuffersInternal: Can't get the shared handle from the dxgiResource.";
+                      setDoingOkay(false);
+                      return false;
+                    }
+                    dxgiResource->Release(); // we don't need this anymore
+
+                                             // Get a pointer on our device to the texture we're getting
+                                             // from the caller's device.
+                    hr = atwDevice->OpenSharedResource(newInfo.sharedResourceHandle, __uuidof(ID3D11Texture2D),
+                      (LPVOID*)&texture2D);
+                    if (FAILED(hr)) {
+                      m_log->error() << "RenderManagerD3D11ATW::"
+                        << "RegisterRenderBuffersInternal: - failed to open shared resource "
+                        << "for buffer " << i;
+                      setDoingOkay(false);
+                      return false;
+                    }
 
                     // We do not need a render target view for the ATW thread -- it will
                     // only be reading from the buffer, not rendering into it.  Our base
                     // class will create our RenderTargetView the first time the app calls
                     // Render().
+                    newInfo.atwBuffer.D3D11 = new osvr::renderkit::RenderBufferD3D11();
+                    newInfo.atwBuffer.D3D11->colorBuffer = texture2D;
                     newInfo.atwBuffer.D3D11->colorBufferView = nullptr; // We don't need this.
-
-					newInfo.atwBuffer.D3D11->depthStencilBuffer = nullptr;
-					newInfo.atwBuffer.D3D11->depthStencilView = nullptr;
-
+		    newInfo.atwBuffer.D3D11->depthStencilBuffer = nullptr;
+		    newInfo.atwBuffer.D3D11->depthStencilView = nullptr;
```

Comment:
Intentional indentation change.  Maybe a tabs-vs-spaces thing, but this was lined up wrong when viewed in my editor.  I see that it left some tabs in there, so have now replaced them and pushed a new commit that uses all spaces to avoid this confusion.